### PR TITLE
Added docker documentation for Expose

### PR DESCRIPTION
**Pull request by rubensfig**
What I did
I added some documentation about the Expose flag in the Dockerfile that allows to expose the protocol associated with a port.

Signed-off-by: Rubens Figueiredo <r.figueiredo.52@gmail.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: docs/reference/builder.md [hunk 1]
```diff
@@ -754,14 +754,15 @@ This will then be visible from `docker inspect` with the other labels.
...
 
 The `EXPOSE` instruction informs Docker that the container listens on the
-specified network ports at runtime. `EXPOSE` does not make the ports of the
-container accessible to the host. To do that, you must use either the `-p` flag
-to publish a range of ports or the `-P` flag to publish all of the exposed
-ports. You can expose one port number and publish it externally under another
-number.
+specified network ports at runtime. You can also specify if the port uses TCP/UDP,
+and defaults to tcp if the protocol is omitted.
+`EXPOSE` does not make the ports of the container accessible to the host.
```
Comment (mstanleyjones) [inline docs/reference/builder.md]: Maybe say something about `EXPOSE` really being a documentation mechanism between the person who builds the image and the person who runs the container.

Comment (rubensfig) [inline docs/reference/builder.md]: I was then thinking this, "`EXPOSE` does not make the ports of the specified network to the host ports at runtime, instead it allows for users running the container to be aware what ports are available in the container". 
Also, I forgot to mention that I made this pull request when I saw this [issue](https://github.com/moby/moby/issues/34373), is there any problem? Sorry, I'm quite new to this.

Comment (mstanleyjones) [inline docs/reference/builder.md]: Nope, no problem. :) Go ahead and do your rewording and I'll commit into this PR with any copyedits that need to be made.

**Comment by mstanleyjones** [comment]
I just squashed your two commits together and added a copyedit commit on top of them. Let me know what you think!

**Comment by rubensfig** [comment]
Ok, that is great! Thank you for the help.

**Pull request merged**